### PR TITLE
Push notifications - Update Settings

### PR DESCRIPTION
**Pull request by noahmalmed**
A few things going on here:
- Complete Firebase integration for topics
	- Add subscription options
	- Add default subscriptions 
- Rewrite Settings with redux persist (this allows us to easily persist user settings)
- Update Settings UI 
- Update dependency name 

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

### CODE REVIEW

File: src/components/settings/Settings.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        <SettingHeader text="General Settings" />
+        <View style={styles.workflowContainer}>
+          <SettingsToggle
+            title="Show all workflows"
+            description="Includes non-native workflows"
+            onToggle={() => {this.props.settingsActions.updateShowAllWorkflow(!this.props.showAllWorkflows)}}
+            value={this.props.showAllWorkflows}
+          />
+        </View>
+        <Separator style={styles.titlePadding} color={'rgba(151, 151, 151, 1)'}/>
```
Comment (wgranger) [inline src/components/settings/Settings.js]: Can this be given a global value or constant? Not sure what 151, 151, 151, 1 converts to.

File: src/containers/app.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ import { PersistGate } from 'redux-persist/integration/react'
 const persistConfig = {
   key: 'root',
   storage,
-  whitelist: ['images', 'user'] // Images and User data will be persisted
+  whitelist: ['images', 'user', 'settings'] // Images and User data will be persisted
```
Comment (wgranger) [inline src/containers/app.js]: Can you update this comment as I assume settings are now persisted.

**Comment by noahmalmed** [comment]
@wgranger changes made per your review

**Comment by wgranger** [comment]
Looks good

**Pull request merged**